src/py/mat3ra/made/tools/build/interface.py

```diff
@@ -0,0 +1,169 @@
+
+        Args:
+            termination (Union[int, tuple]): Termination index or tuple.
+            slice_range (slice): A slice object to specify the range of interfaces to return.
+
+        Returns:
+            List[Interface]: List of interfaces for the termination.
+        """
+        if isinstance(termination, int):
+            termination = self.terminations[termination]
+
+        interfaces = self.interfaces.get(termination, [])
+
+        if slice_range is not None:
+            return interfaces[slice_range]
+
+        return interfaces
+
+    def remove_duplicate_interfaces(self, strain_mode=StrainModes.mean_abs_strain):
+        for termination in self.terminations:
+            self.remove_duplicate_interfaces_for_termination(termination, strain_mode)
+
+    def remove_duplicate_interfaces_for_termination(self, termination, strain_mode=StrainModes.mean_abs_strain):
+        def are_interfaces_duplicate(interface1, interface2):
+            return interface1.num_sites == interface2.num_sites and np.allclose(
+                interface1.interface_properties[strain_mode], interface2.interface_properties[strain_mode]
+            )
+
+        sorted_interfaces = self.get_interfaces_for_termination_sorted_by_size(termination)
+        filtered_interfaces = [sorted_interfaces[0]] if sorted_interfaces else []
+
+        for interface in sorted_interfaces[1:]:
+            if not any(
+                are_interfaces_duplicate(interface, unique_interface) for unique_interface in filtered_interfaces
+            ):
+                filtered_interfaces.append(interface)
+
+        self.set_interfaces_for_termination(termination, filtered_interfaces)
+
+    def get_interfaces_for_termination_sorted_by_strain(self, termination, strain_mode=StrainModes.mean_abs_strain):
+        return sorted(
+            self.get_interfaces_for_termination(termination),
+            key=lambda x: np.mean(np.abs(x.interface_properties[strain_mode])),
+        )
+
+    def get_interfaces_for_termination_sorted_by_size(self, termination):
+        return sorted(
+            self.get_interfaces_for_termination(termination),
+            key=lambda x: x.num_sites,
+        )
+
+    def get_interfaces_for_termination_sorted_by_strain_and_size(
+        self, termination, strain_mode=StrainModes.mean_abs_strain
+    ):
+        return sorted(
+            self.get_interfaces_for_termination_sorted_by_strain(termination, strain_mode),
+            key=lambda x: x.num_sites,
+        )
+
+    def sort_interfaces_for_all_terminations_by_strain_and_size(self):
+        for termination in self.terminations:
+            self.set_interfaces_for_termination(
+                termination, self.get_interfaces_for_termination_sorted_by_strain_and_size(termination)
+            )
+
+    def get_all_interfaces(self):
+        return functools.reduce(lambda a, b: a + b, self.interfaces.values())
+
+    @convert_result_to_material
```

Comment:
Why are we only adding the decorator to this function?? Let's have `get_interfaces_for_termination_at_index` and `get_interfaces_for_termination_at_index_as_materials` for clarity